alert-common/src/main/java/com/synopsys/integration/alert/common/rest/ProxyTestService.java

```diff
@@ -0,0 +1,61 @@
+package com.synopsys.integration.alert.common.rest;
+
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.alert.api.common.model.exception.AlertException;
+import com.synopsys.integration.alert.common.AlertProperties;
+import com.synopsys.integration.alert.common.rest.model.SettingsProxyModel;
+import com.synopsys.integration.alert.common.rest.proxy.ProxyManager;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.Slf4jIntLogger;
+import com.synopsys.integration.rest.HttpUrl;
+import com.synopsys.integration.rest.RestConstants;
+import com.synopsys.integration.rest.client.IntHttpClient;
+import com.synopsys.integration.rest.proxy.ProxyInfo;
+import com.synopsys.integration.rest.request.Request;
+import com.synopsys.integration.rest.response.Response;
+
+//TODO: This should be in the component subproject but currently requires several dependencies not available to that subproject.
+@Component
+public class ProxyTestService {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final AlertProperties alertProperties;
+    private final ProxyManager proxyManager;
+    private final Gson gson;
+
+    @Autowired
+    public ProxyTestService(AlertProperties alertProperties, ProxyManager proxyManager, Gson gson) {
+        this.alertProperties = alertProperties;
+        this.proxyManager = proxyManager;
+        this.gson = gson;
+    }
+
+    public void pingHost(String testUrl, SettingsProxyModel settingsProxyModel) throws IntegrationException {
+        ProxyInfo proxyInfo = proxyManager.createProxyInfo(settingsProxyModel);
+        IntHttpClient client = createIntHttpClient(proxyInfo);
+
+        HttpUrl httpUrl = new HttpUrl(testUrl);
+        Request testRequest = new Request.Builder(httpUrl).build();
+        try (Response response = client.execute(testRequest)) {
+            if (RestConstants.OK_200 >= response.getStatusCode() && response.getStatusCode() < RestConstants.MULT_CHOICE_300) {
+                logger.info("Successfully pinged {}!", testUrl);
+            } else {
+                throw new AlertException(String.format("Could not ping: %s. Status Message: %s. Status code: %s", testUrl, response.getStatusMessage(), response.getStatusCode()));
```

Comment:
If possible we should avoid using exceptions to handle flow. Since this is a new feature we should be able to have this return some result model and if our other new code doesn't allow this we should fix it so that it does.

Reply:
For this we should probably have a ProxyTestResult with a boolean indicating success or failure the status code and status message returned.

Reply:
Digging around, I believe this will also be an issue for email as we are also using exceptions to manage the flow there.